apps/minifront/src/state/ibc.ts

```diff
@@ -90,29 +90,57 @@ export const createIbcSendSlice = (): SliceCreator<IbcSendSlice> => (set, get) =
   };
 };
 
+// Timeout is two days. However, in order to prevent identifying oneself by clock skew,
+// timeout time is rounded to the nearest 10 minute mark.
+// Reference in core: https://github.com/penumbra-zone/penumbra/blob/1376d4b4f47f44bcc82e8bbdf18262942edf461e/crates/bin/pcli/src/command/tx.rs#L1066-L1067
+export const currentTimePlusTwoDaysRounded = (): bigint => {
+  // Get current time in milliseconds and convert to nanoseconds
+  const nowInNanoseconds = BigInt(Date.now()) * BigInt(1_000_000);
+
+  // New timestamp after adding 2 days
+  const twoDays = BigInt(2 * 24 * 60 * 60 * 1000 * 1000000);
+  const nowPlusTwoDays = nowInNanoseconds + twoDays;
+  const tenMinutes = BigInt(10 * 60 * 1000 * 1000000);
+
+  // Round up to nearest 10 minutes
+  return nowPlusTwoDays + tenMinutes - (nowPlusTwoDays % tenMinutes);
+};
+
+// Reference in core: https://github.com/penumbra-zone/penumbra/blob/1376d4b4f47f44bcc82e8bbdf18262942edf461e/crates/bin/pcli/src/command/tx.rs#L998-L1050
 const getTimeout = async (
   chain: Chain,
 ): Promise<{ timeoutTime: bigint; timeoutHeight: Height }> => {
```

Comment:
Turns out there can be many client states with the same chain id. This new method reflects the way core does it: https://github.com/penumbra-zone/penumbra/blob/1376d4b4f47f44bcc82e8bbdf18262942edf461e/crates/bin/pcli/src/command/tx.rs#L998-L1050. It precisely gets the client state we need.